File: 985 - Sum of Even Numbers After Queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: list[int], queries: list[list[int]]) -> list[int]:
        res = []
        evenIdx = set()
        evenSum = 0
        for v in range(len(nums)):
            if nums[v] % 2 == 0:
                evenIdx.add(v)
                evenSum += nums[v]
        for val, i in queries:
            if (nums[i] + val) % 2 == 0:
                evenSum += val
                if i not in evenIdx:
                    evenIdx.add(i)
                    evenSum += nums[i]
            elif i in evenIdx:
                    evenIdx.remove(i)
                    evenSum -= nums[i]
            nums[i] += val
            res.append(evenSum)
        return res

File: 985 - Sum of Even Numbers After Queries.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int[] sumEvenAfterQueries(int[] nums, int[][] queries) {
        Set<Integer> evenIdx = new HashSet<>();
        int evenSum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] % 2 == 0) {
                evenSum += nums[i];
                evenIdx.add(i);
            }
        }
        int[] res = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int val = queries[i][0], idx = queries[i][1];
            if ((nums[idx] + val) % 2 == 0) {
                evenSum += val;
                if (!evenIdx.contains(idx)) {
                    evenSum += nums[idx];
                    evenIdx.add(idx);
                }
            } else {
                if (evenIdx.contains(idx)) {
                    evenSum -= nums[idx];
                    evenIdx.remove(idx);
                }
            }
            nums[idx] += val;
            res[i] = evenSum;
        }
        return res;
    }
}